### ISSUE TITLE
secretKey on frontend?

### ISSUE THREAD

**Issue by chug2k** [Fausto95/aws-s3#67]
The instructions ask you to put your secret key as a parameter. But to my understanding...you probably do not want to publish your secretKey in a client React App. Am I missing something?

**Comment by anthonyalviz** [comment]
use a .env file and add it to your .gitignore file.

**Comment by chug2k** [comment]
Understood that would keep it out of your git repository, but doesn't that still publish your secretKey when you eventually push your source code? 

The build step would replace the output with your secretKey,  and then every client would download the .js file ... with your secret key?

**Comment by jasonwu2153** [comment]
Yeah this is not secure - much better way to accomplish direct upload to AWS is using presigned urls.

**Comment by ghassanmas** [comment]
I agree with @chug2k,
I think the deafult instruction on the readme should explain more about the vulnerability of exposing the keys, thought the [ploiciy used](https://github.com/Fausto95/aws-s3/blob/master/README.md#s3-bucket-policy) stricits requests to site origins only,  still that wouldn't be enoguh. 
AWS has tutrolial of similiar use case,[s3-example-photo-album](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/s3-example-photo-album.html), where they included this warning in the begining: 
> If you enable access for unauthenticated users, you will grant write access to the bucket, and all objects in the bucket, to anyone in the world. This security posture is useful in this example to keep it focused on the primary goals of the example. In many live situations, however, tighter security, such as using authenticated users and object ownership, is highly advisable.

I think simliar message wording should be added to the readme.
